static/app/components/codecov/virtualRenderers/useSyncWrapperWidth.ts

```diff
@@ -0,0 +1,35 @@
+import {useLayoutEffect, useState} from 'react';
+
+/**
+ * This hook gets the width of the wrapper element and syncs it with the width of the
+ * content.
+ * @returns The width of the wrapper element and setter for the wrapper ref.
+ */
+export const useSyncWrapperWidth = () => {
+  const [wrapperWidth, setWrapperWidth] = useState<`${number}px` | '100%'>('100%');
+  const [wrapperRefState, setWrapperRefState] = useState<HTMLDivElement | null>(null);
```

Comment:
Do you find it confusing this has `ref` in it's name but isn't using the `useRef` hook? And confirming this is not meant to be a `useRef` as well

Reply:
Yep we want this naming convention because we use the `ref` prop (function), and assign it to the `wrapperRefState`:

```tsx
    <div className="flex flex-1" ref={setWrapperRefState}>...</div>
```

We want to do this, because we want to trigger a re-render when the ref info changes instead of nothing happening.